src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,132 @@
+package duke.parser;
+
+import java.time.LocalDate;
+
+import duke.command.Command;
+import duke.command.AddCommand;
+import duke.command.DeleteCommand;
+import duke.command.ByeCommand;
+import duke.command.DateCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnMarkCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyDescriptionException;
+import duke.exception.InvalidInputException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+    private final static char UNDONE_STATUS = 'U';
+
+    /**
+     * Parses the user command.
+     *
+     * @param input The input from user.
+     * @return The corresponding command class.
+     * @throws DukeException Throw exception when there is an invalid user input.
+     */
+    public static Command parseCommand(String input) throws DukeException{
+        String[] arguments =  input.trim().split(" ", 2);
+        String commandType = arguments[0];
+            if ("bye".equalsIgnoreCase(commandType)) {
+               return exit();
+            } else if ("list".equalsIgnoreCase(commandType)) {
+                return listAllTasks();
+            } else if ("unmark".equalsIgnoreCase(commandType)) {
+                return unMarkTask(arguments);
+            } else if ("mark".equalsIgnoreCase(commandType)) {
+                 return markTask(arguments);
+            } else if ("todo".equalsIgnoreCase(commandType)) {
+               return addToDo(arguments);
+            } else if ("deadline".equalsIgnoreCase(commandType)) {
+               return addDeadline(arguments);
+            } else if ("event".equalsIgnoreCase(commandType)) {
+              return addEvent(arguments);
+            } else if ("date".equalsIgnoreCase(commandType)) {
+               return listTasksOnDate(arguments);
+            } else if ("delete".equalsIgnoreCase(commandType)) {
+               return deleteTask(arguments);
+            } else {
+                throw new InvalidInputException();
+            }
```

Comment:
There's an unnecessary extra layer of indentation here. On a side note, is there a reason why you used this instead of a switch case? It might help clean up the code a little

Reply:
Thanks for pointing that out, I am using this because I want to make the commands non-case sensitive by using the equalsIgnoreCase function.